commit 5ac9e85a213017322e0fbdc604a9fc26f157a815
Author: Volker Theile <vtheile@suse.com>
Date:   Fri Jan 25 14:45:01 2019 +0100

    mgr: Improve ActivePyModules::get_typed_config implementation
    
    - Remove PyModuleRegistry::module_exists method
    - Refactor PyModuleRegistry::get_module method
    
    Signed-off-by: Volker Theile <vtheile@suse.com>

diff --git a/src/mgr/ActivePyModules.cc b/src/mgr/ActivePyModules.cc
index 1f185714972..06b478c4ec1 100644
--- a/src/mgr/ActivePyModules.cc
+++ b/src/mgr/ActivePyModules.cc
@@ -529,15 +529,15 @@ PyObject *ActivePyModules::get_typed_config(
   const std::string &module_name,
   const std::string &key) const
 {
-  if (!py_module_registry.module_exists(module_name)) {
-    derr << "Module '" << module_name << "' is not available" << dendl;
-    Py_RETURN_NONE;
-  }
-
   std::string value;
   bool found = get_config(module_name, key, &value);
   if (found) {
     PyModuleRef module = py_module_registry.get_module(module_name);
+    if (!module) {
+        derr << "Module '" << module_name << "' is not available" << dendl;
+        Py_RETURN_NONE;
+    }
+
     dout(10) << __func__ << " " << key << " found: " << value << dendl;
     return module->get_typed_option_value(key, value);
   }
